### PR TITLE
Summarize ForestIndex into a two-blob summary

### PR DESCRIPTION
**Pull request by noencke**

### CODE REVIEW

File: packages/dds/tree/src/feature-libraries/forestIndex.ts [hunk 1]
```diff
@@ -33,14 +38,13 @@ import { jsonableTreeFromCursor } from "./treeTextCursor";
...
-    // TODO: implement this to provide snapshots in summaries.
     readonly summaryElement?: SummaryElement = this;
 
     private readonly cursor: ITreeSubscriptionCursor;
 
     // Note that if invalidation happens when these promises are running, you may get stale results.
-    private readonly treeBlob: ICachedValue<Promise<ISummaryAttachment>>;
-    private readonly schemaBlob: ICachedValue<Promise<ISummaryAttachment>>;
+    private readonly treeBlob: ICachedValue<Promise<IFluidHandle<ArrayBufferLike>>>;
+    private readonly schemaBlob: ICachedValue<Promise<IFluidHandle<ArrayBufferLike>>>;
```
Comment (jenn-le) [inline packages/dds/tree/src/feature-libraries/forestIndex.ts:47]: This is probably a @CraigMacomber question, but what kind of schema data is stored in a ForestIndex?

File: packages/dds/tree/src/feature-libraries/forestIndex.ts [hunk 2]
```diff
@@ -105,78 +107,60 @@ export class ForestIndex implements Index<unknown>, SummaryElement {
     }
 
     getAttachSummary(
```
Comment (jenn-le) [inline packages/dds/tree/src/feature-libraries/forestIndex.ts:109]: Is it worth sharing some code between this and `summarize`?

File: packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts [hunk 3]
```diff
@@ -134,24 +137,50 @@ export interface SummaryElement {
      */
     readonly key: string;
 
-    // See IChannel
+    /**
+     * {@inheritDoc @fluidframework/datastore-definitions#(IChannel:interface).getAttachSummary}
+     * @param stringify - Serializes the contents of the index (including {@link IFluidHandle}s) for storage.
+     */
     getAttachSummary(
+        stringify: SummaryElementStringifier,
```
Comment (jenn-le) [inline packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts:145]: The implementation param name is `serialize`, we should probably pick one. I'm also wondering if `SummaryElementSerializer` would be a better name, `Serializer` seems to be the standard in the repo.

Comment (jenn-le) [reply packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts:145]: Same for summarize.

Comment (noencke) [reply packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts:145]: Oops, good catch. I think they should both be `stringify`. `Serializer` is indeed a standard already, and it means something else. The function _on_ `Serializer`, which looks very much like this one, is called `stringify`.

File: packages/dds/tree/src/feature-libraries/forestIndex.ts (hunk 2 above)
Comment (noencke) [reply packages/dds/tree/src/feature-libraries/forestIndex.ts:109]: Yea that's a cool idea. A function which takes the two strings and returns the summary tree. Let's call it... `summarizeCore` :D

**Review by jenn-le — APPROVED** [review]

**Pull request merged**